src/varity/ref_gene.clj

```diff
@@ -206,29 +269,32 @@
                   :tx-end (:end transcript)
                   :strand strand))))
 
-(defn load-gencode
-  [f parse-line & {:keys [chunk-size] :or {chunk-size 10000}}]
+(defn load-gtf-gff
+  [f build-feature-map attr-kv-sep & {:keys [chunk-size] :or {chunk-size 10000}}]
   (with-open [rdr (io/reader (util/compressor-input-stream f))]
-    (let [feature-map (->> (line-seq rdr)
+    (let [parse-line-fn #(parse-gtf-gff-line % :attr-kv-sep attr-kv-sep)
+          feature-map (->> (line-seq rdr)
                            (partition-all chunk-size)
-                           (pmap #(doall (keep parse-line %)))
+                           (pmap #(doall (keep parse-line-fn %)))
                            (apply concat)
                            build-feature-map)]
       (doall (map (partial ->region feature-map) (:transcript feature-map))))))
 
-(defn load-gtf
-  [f & {:keys [chunk-size] :or {chunk-size 10000} :as opts}]
+(defn load-gencode
+  [f & {:keys [file-type chunk-size] :or {file-type :gtf chunk-size 10000} :as opts}]
   (let [args (concat [f
-                      parse-gencode-line]
+                      build-gencode-feature-map
+                      (file-type gtf-gff-attr-kv-sep)]
                      (mapcat seq opts))]
-    (apply load-gencode args)))
+    (apply load-gtf-gff args)))
 
-(defn load-gff3
-  [f & {:keys [chunk-size] :or {chunk-size 10000} :as opts}]
+(defn load-ncbi-genome-annotation
+  [f & {:keys [file-type chunk-size] :or {file-type :genome-annotation-gtf chunk-size 10000} :as opts}]
```

Comment:
Can we support `:file-type :gff3`? If not, that keyword arg is confusing. It looks like caller can choose it.